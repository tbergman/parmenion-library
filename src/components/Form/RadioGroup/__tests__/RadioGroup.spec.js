/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { RadioGroup, Radio } from '../../';

it('renders correctly', () => {
  const tree = shallow(
    <RadioGroup>
      <Radio value="1">Option 1</Radio>
      <Radio value="2">Option 2</Radio>
      <Radio value="3">Option 3</Radio>
    </RadioGroup>,
  );

  expect(tree).toMatchSnapshot();
});

it('should throw an error when a child isn\'t a <Radio />', () => {
  const tree = () => shallow(
    <RadioGroup>
      <Radio value="1">Option 1</Radio>
      <div>Option 2</div>
      <Radio value="3">Option 3</Radio>
    </RadioGroup>,
  );

  expect(tree).toThrowError('Unexpected component in <RadioGroup />, make sure all children are <Radio /> components.');
});
