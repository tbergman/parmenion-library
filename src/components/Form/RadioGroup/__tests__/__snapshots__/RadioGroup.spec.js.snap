exports[`test renders correctly 1`] = `
<Horizontal
  spaceBetween={1}>
  <Radio
    checked={false}
    isInline={false}
    isSmall={false}
    status={0}
    value="1">
    Option 1
  </Radio>
  <Radio
    checked={false}
    isInline={false}
    isSmall={false}
    status={0}
    value="2">
    Option 2
  </Radio>
  <Radio
    checked={false}
    isInline={false}
    isSmall={false}
    status={0}
    value="3">
    Option 3
  </Radio>
</Horizontal>
`;
